modules/460-log-shipper/hooks/internal/vector/vrl/cleanup.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vrl
+
+// CleanUpRule is a general cleanup rule to sanitize the final message.
+// It should always be the first rule in the transforms chain.
+const CleanUpRule Rule = `
```

Comment:
We enrich logs with some kubernetes metadata and then cleanup parts of this metadata. I hope I'm wrong

Reply:
I am afraid that I did not get this question right.

Before transforming, we use this cleanup rule to avoid leaking data from the source, e.g., unwanted [field annotations](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#pod_annotation_fields) to transforms.